javalib/src/main/scala/java/util/StringTokenizer.scala

```diff
@@ -1,103 +1,93 @@
 package java.util
 
-class StringTokenizer(string: String,
-                      private var delimiters: String,
-                      returnDelimiters: Boolean)
-    extends java.util.Enumeration[Object] {
+class StringTokenizer(
+  str: String,
+  private var delim: String,
+  returnDelims: Boolean
+) extends java.util.Enumeration[Object] {
 
-  def this(string: String) = this(string, " \t\n\r\f", false)
-  def this(string: String, delimeters: String) =
-    this(string, delimeters, false)
+  def this(str: String) = this(str, " \t\n\r\f", false)
+  def this(str: String, delim: String) = this(str, delim, false)
 
-  if (string == null) {
-    throw new NullPointerException()
-  }
-
-  private var position = 0
+  private var position: Int = 0
+  private val length: Int = str.length
 
   def countTokens(): Int = {
-    var count   = 0
-    var inToken = false
-    val length  = string.length
-    for (i <- position until length) {
-      if (delimiters.indexOf(string.charAt(i), 0) >= 0) {
-        if (returnDelimiters)
+    var count: Int = 0
+    var inToken: Boolean = false
+    var i: Int = position
+
+    while(i < length) {
```

Comment:
So, Am I following along correctly that the major benefit of this PR is 
changing known dog slow Scala for loops over to while loops?

It looks to me like this is one of the for-over-a-range loops that was optimized
in Scala Native (0.4.0-M2 or earlier?).   If this is true, I would have to step back out to look at the entire file to see if changing to a while is still a simplification or uses
a better algorithm.